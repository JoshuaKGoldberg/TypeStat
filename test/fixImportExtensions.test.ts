import path from "node:path";
import { expect, test } from "vitest";

import { runMutationTest } from "../src/tests/testSetup.js";

test("import extensions", async () => {
	const caseDir = path.join(
		import.meta.dirname,
		"cases/fixes/importExtensions",
	);
	const { actualContent, expectedFilePath } = await runMutationTest(caseDir);
	await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
});
