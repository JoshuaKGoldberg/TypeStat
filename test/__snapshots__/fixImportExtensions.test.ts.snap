// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`import extensions > options 1`] = `
"[
  {
    "cleanups": {
      "suppressTypeErrors": true
    },
    "compilerOptions": {
      "noImplicitAny": false,
      "noImplicitThis": false,
      "strictNullChecks": false
    },
    "files": {
      "above": "",
      "below": "",
      "renameExtensions": false
    },
    "filters": [],
    "fixes": {
      "importExtensions": true,
      "incompleteTypes": false,
      "missingProperties": false,
      "noImplicitAny": false,
      "noImplicitThis": false,
      "noInferableTypes": false,
      "strictNonNullAssertions": false
    },
    "hints": {
      "react": {
        "propTypes": "whenRequired",
        "propTypesOptionality": "asWritten"
      }
    },
    "mutators": [
      [
        "fixImportExtensions",
        null
      ],
      [
        "fixIncompleteTypes",
        null
      ],
      [
        "fixMissingProperties",
        null
      ],
      [
        "fixNoImplicitAny",
        null
      ],
      [
        "fixNoImplicitThis",
        null
      ],
      [
        "fixNoInferableTypes",
        null
      ],
      [
        "fixStrictNonNullAssertions",
        null
      ]
    ],
    "output": {},
    "package": {
      "directory": "<rootDir>",
      "file": "<rootDir>/package.json"
    },
    "postProcess": {
      "shell": []
    },
    "projectPath": "<rootDir>/tsconfig.json",
    "types": {
      "strictNullChecks": false
    }
  }
]"
`;
